src/lib/test/ApplicationPathHelpers_GTest.cpp

```diff
@@ -0,0 +1,33 @@
+/***********************************************************************************************************************
+*  OpenStudio(R), Copyright (c) Alliance for Sustainable Energy, LLC.
+*  See also https://openstudio.net/license
+***********************************************************************************************************************/
+
+#include <gtest/gtest.h>
+
+#include <utilities/core/ApplicationPathHelpers.hpp>
+#include <utilities/core/PathHelpers.hpp>
+
+using namespace openstudio;
+
+// Want to make sure what getOpenStudioModule actually returns
+TEST(ApplicationPathHelpers, Simple_test_forThisModule) {
+  path openstudioModulePath = getOpenStudioModule();
+  EXPECT_TRUE(exists(openstudioModulePath));
+  // The expected path is the utilities one, but resolved for symlinks (we don't want to hardcode the version eg openstudio_utilities_tests-2.8.0)
+#if defined(_WIN32)
+#  if defined(NINJA)
+  openstudio::path expectedOpenstudioModulePath = getApplicationBuildDirectory() / toPath("Products/openstudiolib.dll");
+#  elif _DEBUG
+  openstudio::path expectedOpenstudioModulePath = getApplicationBuildDirectory() / toPath("Products/Debug/openstudiolib.dll");
+#  else
+  openstudio::path expectedOpenstudioModulePath = getApplicationBuildDirectory() / toPath("Products/Release/openstudiolib.dll");
+#  endif
+#elif __APPLE__
+  openstudio::path expectedOpenstudioModulePath = getApplicationBuildDirectory() / toPath("Products/libopenstudiolib.dylib");
+#else
+  openstudio::path expectedOpenstudioModulePath = getApplicationBuildDirectory() / toPath("Products/libopenstudiolib.so");
+#endif
+  expectedOpenstudioModulePath = completeAndNormalize(expectedOpenstudioModulePath);
+  EXPECT_EQ(toString(expectedOpenstudioModulePath), toString(openstudioModulePath));
+}
```

Comment:
I wanted to keep the test with openstudiolib.so though. Instead of strapping it in openstudio_model_tests I created a new test exe... not sure if wise. @kbenne please

Reply:
This kind of stuff is a pain. So many configurations and contexts to consider. I think it seems fine to create a new test executable. 

These tests assert truth in the dev/testing environment. Sadly it doesn't assert truth in the end user installed situation, so the value of seems less than ideal. Also, with the amount of logic for the various build configs it seems a bit brittle. Having said that, at least we'll know if it breaks. :)